### PR TITLE
[EFR32 examples] Move EFR platform files higher up the tree for other examples uses

### LINKED ISSUE #2415: EFR32 lock_app doesn't have a factory reset
#### Problem
Factory reset not available on EFR32 lock_app. You have to manually change some openthread network info with cli commands if you want to change network or redo the paring.

 #### Proposed Solution
Reuse PB0 to trigger the factory Reset too after some time and reset the configuration manager

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
Some EFR board files are generic to EFR projects and could be reused for other EFR examples.
Factory reset is not available 

 #### Summary of Changes
Move EFR32 board specific files in examples platform so they can be reused for other demos
Use those board files for opentread gn build so we add or remove board supports from the CHIP repo independantly
Add trigger in the function button handler to have Start joiner, Factory Reset and Software Update on the same button
Complete factory reset functionality to erase Chip configs and Thread network provison 

fixes #2415

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
>  Complete factory reset functionality for

I'm not sure what this part of the PR description means.  Could that be fixed, please?

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by jmartinez-silabs** [comment]
@bzbarsky-apple 
> > Complete factory reset functionality for
> 
> I'm not sure what this part of the PR description means. Could that be fixed, please?

Updated

**Comment by rwalker-apple** [comment]
@hawk248 @jelderton @mspang ?

File: examples/lock-app/efr32/BUILD.gn [hunk 1]
```diff
@@ -63,6 +64,7 @@ efr32_executable("lock_app") {
   include_dirs += [
     "${efr32_project_dir}/include/",
     "${chip_root}/src/app/util",
+    "${examples_plat_dir}/",
```
Comment (mspang) [inline examples/lock-app/efr32/BUILD.gn:67]: nit: remove this slash

**Review by mspang — APPROVED** [review]

**Pull request merged**